src/validation/validationHealthcareProfessional.ts

```diff
@@ -0,0 +1,134 @@
+import * as gqlTypes from '../typeDefs/gqlTypes.js'
+import { isInvalidName, hasJapaneseCharacters, hasLatinCharacters } from '../../utils/stringUtils.js'
+import { ErrorCode, Result } from '../result.js'
+
+export default function validateNames(
+    names: gqlTypes.LocalizedNameInput[] | undefined | null,
+    validationResults: Result<unknown>
+): void {
+    if (!names || !names?.length) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+        return
+    }
+    
+    if (names && names.length > 16) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    const localesUsed: string[] = []
+
+    names?.forEach((nameObject, index) => {
+        const isLocalUsed: boolean = localesUsed.includes(nameObject.locale)
+
+        if (isLocalUsed) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `names[${index}].locale`,
+                errorCode: ErrorCode.CONTAINS_DUPLICATE_LOCAL,
+                httpStatus: 400
+            })
+        } else {
+            localesUsed.push(nameObject.locale)
+        }
+        
+        if (nameObject.firstName.length > 30) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: `names[${index}].firstName`,
+                errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+                httpStatus: 400
+            })
+        }
+    
+        if (!nameObject.firstName.trim()) {
```

Comment:
should check for null before calling trim since we haven't checked firstname before this. 

```suggestion
        if (!nameObject.firstName?.trim()) {
```